chain/network/src/peer_manager/peer_store/mod.rs

```diff
@@ -290,8 +322,16 @@ impl PeerStore {
                 Entry::Vacant(entry) => entry,
             };
             entry.insert(VerifiedPeer::signed(peer_info.id.clone()));
-            peerid_2_state
-                .insert(peer_info.id.clone(), KnownPeerState::new(peer_info.clone(), now));
+
+            if let Some((_, popped_peer_state)) = peerid_2_state
+                .push(peer_info.id.clone(), KnownPeerState::new(peer_info.clone(), now))
+            {
+                // If a peer was evicted from peer_states due to the bounded cache size
```

Comment:
this logic is duplicated: either wrap addr_peers and peer_states into a separate data structure which supports update/insert operation, or get rid of one of these (see previous comment).